examples/web/cache/refresh-disposable-tokens.ts

```diff
@@ -0,0 +1,250 @@
+  console.log(`Generated a disposable token that will expire at ${humanReadableTime}`);
+  const disposableToken = {
+    token: tokenResponse.authToken,
+    expiresAt: tokenResponse.expiresAt,
+  };
+  return disposableToken;
+}
+
+async function tokenVendingMachine(): Promise<{token: string; expiresAt: ExpiresAt}> {
+  const resp = await fetch(process.env.TVM_ENDPOINT as string);
+  const respJson = (await resp.json()) as {authToken: string; expiresAt: number};
+  const disposableToken = {
+    token: respJson.authToken,
+    expiresAt: ExpiresAt.fromEpoch(respJson.expiresAt),
+  };
+  return disposableToken;
+}
+
+interface TokenRefreshingTopicClientProps {
+  refreshBeforeExpiryMs: number;
+  getDisposableToken: () => Promise<{token: string; expiresAt: ExpiresAt}>;
+}
+
+class TokenRefreshingTopicClient {
+  topicClient: TopicClient;
+  refreshBeforeExpiryMs: number;
+  getDisposableToken: () => Promise<{token: string; expiresAt: ExpiresAt}>;
+  activeSubscriptions: Record<
+    string,
+    {
+      cacheName: string;
+      topicName: string;
+      lastSequenceNumber: number;
+      unsubscribe: () => void;
+      onItem: (item: TopicItem) => void;
+      onError: (error: TopicSubscribe.Error) => void;
+    }
+  > = {};
+
+  private constructor(props: TokenRefreshingTopicClientProps) {
+    this.refreshBeforeExpiryMs = props.refreshBeforeExpiryMs;
+    this.getDisposableToken = props.getDisposableToken;
+  }
+
+  private async initialize() {
+    const disposableToken = await this.getDisposableToken();
+    this.topicClient = new TopicClient({
+      credentialProvider: CredentialProvider.fromString(disposableToken.token),
+    });
+    // eslint-disable-next-line @typescript-eslint/no-misused-promises
+    setTimeout(async () => {
+      await this.refreshToken();
+    }, getRefreshAfterMs(disposableToken.expiresAt, this.refreshBeforeExpiryMs));
+    console.log('Initialized topic client and set first timeout');
+  }
+
+  static async create(props: TokenRefreshingTopicClientProps) {
+    const client = new TokenRefreshingTopicClient(props);
+    await client.initialize();
+    return client;
+  }
+
+  private async refreshToken() {
+    console.log('Disposable token expiring soon, refreshing topic client with new token');
+    const disposableToken = await this.getDisposableToken();
+    const newTopicClient = new TopicClient({
+      credentialProvider: CredentialProvider.fromString(disposableToken.token),
+    });
+    // eslint-disable-next-line @typescript-eslint/no-misused-promises
+    setTimeout(async () => {
+      await this.refreshToken();
+    }, getRefreshAfterMs(disposableToken.expiresAt, this.refreshBeforeExpiryMs));
+
+    // for each active subscription, unsubscribe from the old topic client and subscribe to the new one
+    for (const key in this.activeSubscriptions) {
+      const value = this.activeSubscriptions[key];
+      const newSubscription = await newTopicClient.subscribe(value.cacheName, value.topicName, {
```

Comment:
@anitarua I think maybe you need to call your local `subscribe` function here so that you get the wrapped version, don't you?